docs/iam-usage.rst

```diff
@@ -0,0 +1,148 @@
+Using IAM For Users
+===================
+
+.. note:: This document assumes basic knowledge of Google Cloud IAM,
+   see `the docs <https://cloud.google.com/iam/docs/>`_ for details
+
+Resources that implement the IAM interface provide the following methods:
+
+Policy Editing Convenience Methods
+----------------------------------
+
+Common keyword arguments:
+
+- ``retries`` number of times to retry in the
+   even of an interrupted transaction, (not available in ``set_policy``)
+- ``version`` version number of the new policy. Not used for transactionality.
+   provided for end-user use. Left blank, will increment the version by 1.
+
+
+>>> resource.add_role(iam.user('alice@example.com'), iam.roles.OWNER.name)
+True
+
+The first argument is one of the IAM "Member" types:
+
+- ``iam.user(email)`` an individual Google account
+- ``iam.service_account(email)`` a Google Cloud Service Account
+- ``iam.group(email)`` a Google group.
+- ``iam.domain(domain_name)`` a Google apps domain
+- ``iam.ALL_AUTHENTICATED_USERS`` any authenticated user
+- ``iam.ALL_USERS``
+
+.. note:: that all of these are convenience wrappers around strings.
+   See the list of member string formats `here <https://cloud.google.com/iam/docs/managing-policies>`_.
+
+The second argument is a role to added to the specified member. Roles are simply strings that correspond to sets of permissions.
+The ``iam`` module provides a (possibly incomplete) list of curated roles.
+You can acquire the complete list by running ``gcloud iam list-grantable-roles //fully/qualified/resource/path``, and use the
+names provided there in place of ``iam.roles.XXXX.name`` wherever the latter is used in this doc.
+
+This method returns ``True`` if the change to the resource's policy was made successfully, and returns ``False`` otherwise
+(such as in the case of an update being made during this update).
+
+>>> resource.remove_role(iam.user('bob@example.com'), iam.roles.OWNER.name)
+True
+
+
+>>> resource.add_roles(iam.user('alice@example.com'), [iam.roles.OWNER.name, iam.roles.EDITOR.name])
+True
+
+Same as ``resource.add_role`` the second argument is a list of roles to be added to the specified member.
+
+>>> resource.remove_roles(iam.group('devs@example.com'), [iam.roles.OWNER.name, iam.roles.EDITOR.name])
+False
+
+Same as above, but removes the specified roles from the specified member.
+
+>>> resource.add_members(iam.roles.OWNER.name, [iam.domain('example.com'), iam.service_account('compute@iam.my-project.example.com')])
```

Comment:
So these methods accomplish different things. 

If we want there truly to be only one way to do X we would remove every convenience method and only surface the "PolicyChange" interface. I think that's clunky, hence the convenience methods. What I don't want users to do is see `resource.add_roles` and want to add a single role to a bunch of members and do

```
for member in members:
   resource.add_roles(member, [role])
```

Since they won't have a good level of transactionality (and they will make N API requests).